SonarRunner.Shim/PropertiesFileGenerator.cs

```diff
@@ -214,10 +214,18 @@ private static string TryGetCodeCoverageReport(ProjectInfo project, ILogger logg
             return vsCoverageReport;
         }
 
-        private static AnalysisProperties GetAnalysisProperties(AnalysisConfig config, ILogger logger)
+        /// <summary>
+        /// Returns all of the analysis properties that should
+        /// be written to the sonar-project properties file
+        /// </summary>
+        private static AnalysisProperties GetAnalysisPropertiesToWrite(AnalysisConfig config, ILogger logger)
         {
-            AnalysisProperties properties = config.LocalSettings ?? new AnalysisProperties();
+            AnalysisProperties properties = new AnalysisProperties();
 
+            properties.AddRange(config.GetAnalysisSettings(false).GetAllProperties()
+                // Strip out any sensitive properties
+                .Where(p => !p.ContainsSensitiveData()));
+            
```

Comment:
This is the main change for this part. The extension method returns all of the properties, however they were supplied i.e. command line or file. The sensitive properties are filtered out.